dmlambda/handler.py

```diff
@@ -136,10 +139,22 @@ def process_email_handler(event, context):
             # Found the charity, parse
             data = charity_class.parse_email()
 
+            # Validate this is a new donation
+            donation_table = DynamoTable('donations')
+            existing_receipts = donation_table.query_hash("receipt_id", data["receipt_id"],
+                                                          index="ReceiptIndex", limit=10)
+
+            if existing_receipts:
+                # This receipt already exists!
+                print("**** Duplicate receipt detected - Bucket: {} - Key: {} ****".format(bucket, key))
+                # Notify user we didn't process it
+                send_email(charity_class.from_email, "Donatemates: Unable to process donation",
+                           "Sorry, we weren't able to process your donation as you've already submitted it to Donatemates for a match campaign. If you think this was an error, please forward this email to help@donatemates.com and we'll look into it. Thanks!")
```

Comment:
ah. Good point. So could add to the bottom of the email something like:

request_id:<campaign_id>/<receipt_id>/<email_key>

That way it's a little cryptic so not really giving info away, but we could still go lookup their email and know what they were trying to do?